design/design-doc-smaller-providers.md

```diff
@@ -0,0 +1,704 @@
+* Does not achieve a perfect 1:1 ratio of installed-to-used CRDs.
+* The average deployment may require more compute resources, short term.
+
+I'd like to expand on the first pro a little. No new concepts or configuration
+required. To me, this is the key benefit of the proposed approach, and the
+reason I believe it's the best option for our project and community, long term.
+
+Imagine you're learning how Crossplane works for the first time. The first two
+steps on your getting started checklist are probably:
+
+1. Install Crossplane.
+2. Install the Provider(s) for the things you want to manage.
+
+Keep in mind today that the things you might want to manage might be a cloud
+provider like AWS, but it might also be Helm charts, SQL databases, etc. Most
+folks today are running more than one provider.
+
+Under this proposal, this does not change. You'd explain how to get started with
+Crossplane in exactly the same way. The only difference is that in some cases
+you'd think about "the things you want to manage" in a more granular way. You'd
+think "I want RDS, ElastiCache, CloudSQL, GKE, and Helm" not "I want AWS, GCP,
+and Helm".
+
+I would argue that most folks already think this way. Folks don't think "I want
+AWS support" without having some idea which AWS _services_ they want to use.
+Confirming "Does it support AWS?" is a tiny speed bump before you confirm "Does
+it support RDS?" (or whatever services you need).
+
+### Community Impact
+
+Based on a survey of ~40 community members, the average deployment would:
+
+* Install ~100 provider CRDs, down from over 900 today.
+* Run ~9 provider pods, up from ~2 today.
+
+Keep in mind this represents where folks are at with Crossplane today. Over time
+we expect these numbers to grow as production deployments mature.
+
+### Large Services
+
+One service API group from each of the three major providers is quite large:
+
+* `ec2.aws.upbound.io` (94 CRDs)
+* `compute.gcp.upbound.io` (88 CRDs)
+* `networking.azure.upbound.io` (100 CRDs)
+
+`datafactory.azure.upbound.io` is also quite large at 45 CRDs. All other
+services currently have 25 or fewer CRDs.
+
+These large services contain types that are very commonly used. For example
+`ec2.aws.upbound.io` contains the `VPC` and `SecurityGroup` types, which are
+dependencies of many other services. To deploy an EKS cluster you may want to
+create and reference a `VPC` and a `SecurityGroup` (or three). To do so, you
+would need to bring in `provider-aws-ec2` and thus install ~90 superfluous CRDs.
+
+I propose that in these cases where a service:
+
+* Is large, containing 25 or more CRDs
+* Is commonly used as a dependency of other services
+* Contains a few 'core' types alongside many less commonly used types
+
+We break the service up into two providers; "core" and "everything else". For
+example in the case of `ec2.aws.upbound.io` we would create:
+
+* `upbound/provider-aws-ec2-core` - Contains the ~10-20 most commonly used types
```

Comment:
@ulucinar fyi. 
We should keep this in mind while implementing the tooling for code generation, as this would be an exception to the **group by service** logic.

Reply:
Just FYI, @bassam has some thoughts about this. He might suggest we hold off for now - I'm chatting to him today to learn more.

Reply:
I think we should hold off on `Large Services` for now. Service scoped providers are natural grouping and I think any further breakdown starts to impact the user experience and is something we would only be doing for short term performance optimization.

Reply:
> I think we should hold off

I agree after our discussion yesterday. I'll move it to a "potential future work" section. Note that this will change the projected average number of CRDs based on our survey of community members - I think it will go from ~100 to ~180 or so.